dapp/src/components/page/governance/ProposalForm.tsx

```diff
@@ -181,12 +206,25 @@ const ProposalForm: React.FC = () => {
 
       const directoryCid = await client.uploadDirectory(files);
 
-      setIsLoading(false);
-      alert(
-        `Proposal submitted successfully! CID: ${getIpfsBasicLink(directoryCid.toString())}`,
+      const { createProposal } = await import("@service/WriteContractService");
+
+      const proposalId = await createProposal(
+        projectName,
+        proposalName,
+        directoryCid.toString(),
+        Date.now() + 86400000 * import.meta.env.PUBLIC_VOTING_PERIOD,
```

Comment:
This needs to be configurable from the user side

Reply:
got it